library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphTest.java

```diff
@@ -118,35 +185,121 @@ public void shouldThrowUnsupportedExceptionForComputeWithClass() {
     public void shouldThrowUnsupportedExceptionForTx() {
         // Given
         final Graph gafferGraph = getGafferGraph();
-        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION, gafferGraph);
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
 
         // When / Then
         assertThatExceptionOfType(UnsupportedOperationException.class).isThrownBy(() -> graph.tx());
     }
 
+    @Test
+    public void shouldThrowIllegalArgumentExceptionForNoVertexLabel() {
+        // Given
+        final Graph gafferGraph = getGafferGraph();
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
+
+        //Then
+        assertThatExceptionOfType(IllegalArgumentException.class)
+            .isThrownBy(() -> graph.addVertex(T.id, VERTEX_1))
+            .withMessageMatching("Label is required");
+    }
+
+    @Test
+    public void shouldThrowIllegalArgumentExceptionForNoVertexId() {
+        // Given
+        final Graph gafferGraph = getGafferGraph();
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
+
+        //Then
+        assertThatExceptionOfType(IllegalArgumentException.class)
+            .isThrownBy(() -> graph.addVertex(T.label, SOFTWARE_NAME_GROUP))
+            .withMessageMatching("ID is required");
+    }
+
     @Test
     public void shouldAddAndGetVertex() {
         // Given
         final Graph gafferGraph = getGafferGraph();
-        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION, gafferGraph);
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
 
         // When
         graph.addVertex(T.label, SOFTWARE_NAME_GROUP, T.id, VERTEX_1, NAME_PROPERTY, "GafferPop");
         final Iterator<GafferPopVertex> vertices = graph.vertices(Arrays.asList(VERTEX_1, VERTEX_2), SOFTWARE_NAME_GROUP);
 
+        // Then
+        final GafferPopVertex vertex = vertices.next();
+        assertFalse(vertices.hasNext());
```

Comment:
Shouldn't need to use JUnit assert here:
```suggestion
        assertThat(vertices.hasNext()).isFalse();
```
Shame there's not an assert like this: `assertThat(item).doesNotHaveNext()`. Might be an idea for assertJ?

Reply:
@GCHQDeveloper314 what about this?
```suggestion
        assertThat(vertices).isExhausted();
```

Reply:
That's exactly what I was hoping for. I'd forgotten the method had that name. This should be used, although maybe changing all of these can be done in the refactor of this class and not in this PR to keep thing in scope of the original issue.